### PR TITLE
Add reversible gc test

### PR DESCRIPTION
**Pull request by rht**
To check #1515

### CODE REVIEW

File: test/sharness/t0080-repo.sh [hunk 1]
```diff
@@ -44,6 +44,16 @@ test_expect_success "'ipfs pin rm' output looks good" '
 	test_cmp expected1 actual1
 '
 
+test_expect_failure "ipfs repo gc fully reverse ipfs add" '
+    random 100000 41 >gcfile &&
+    du -sb "$IPFS_PATH/blocks" | awk "{print \$1}" >expected &&
```
Comment (rht) [inline test/sharness/t0080-repo.sh]: TODO: use https://github.com/ipfs/go-ipfs/pull/1495/files#diff-5f6c4deecc0549c8bc2129dd8ee569a9R318

File: test/sharness/lib/test-lib.sh [hunk 2]
```diff
@@ -314,3 +314,19 @@ test_str_contains() {
...
+            ;;
+        FreeBSD)
+            DU="du -s -A -B 1"
+            ;;
+        Darwin | DragonFly)
+            DU="du"
+            ;;
+    esac
+        $DU "$1" | awk "{print \$1}"
+}
```
Comment (jbenet) [inline test/sharness/lib/test-lib.sh]: maybe we can make a go-du that's the same everywhere eventually?

Comment (jbenet) [inline test/sharness/lib/test-lib.sh]: - https://github.com/aisola/go-coreutils/
- https://github.com/aisola/go-coreutils/issues/50

**Comment by jbenet** [comment]
pinning keeps failing... @whyrusleeping any idea why that pinning test keeps failing?

**Comment by jbenet** [comment]
thanks @rht this LGTM. i reran the tests. hopefully will pass now. else, looks unrelated. i'll merge

File: test/sharness/lib/test-lib.sh (hunk 2 above)
Comment (rht) [inline test/sharness/lib/test-lib.sh]: Didn't know that a 'go-coreutils' exists.

Comment (rht) [inline test/sharness/lib/test-lib.sh]: But in os x, to get a gnu coreutils, `brew install coreutils`.

**Comment by whyrusleeping** [comment]
https://travis-ci.org/ipfs/go-ipfs/jobs/72941577

Those test failures are new. They arent the pinning ones i'm debugging

**Pull request merged**